### DISCUSSION TITLE
Cara memperbaiki eror untuk menghitung R2. MAE dll nya menggunakan code ini?

### DISCUSSION THREAD

**Discussion by alexanderbudiman** [teamalgoritma/community#731] [Q&A] [answered]
prediction1 %>% 
  summarise(
    R_SQUARED = rsq_vec(strength, .pred),
    RMSE = rmse_vec(strength, .pred),
    MAE = mae_vec(strength, .pred),
    MAPE = mape_vec(strength, .pred),
    MASE = mase_vec(strength, .pred)
  )

Error in `summarise()`:
i In argument: `R_SQUARED =
  rsq_vec(strength, .pred)`.
Caused by error in `rsq_vec()`:
! could not find function "rsq_vec"
Backtrace:
  1. revert_prediction1 %>% ...
  3. dplyr:::summarise.data.frame(...)
  4. dplyr:::summarise_cols(.data, dplyr_quosures(...), by, "summarise")
  6. dplyr:::map(quosures, summarise_eval_one, mask = mask)
  7. base::lapply(.x, .f, ...)
  8. dplyr (local) FUN(X[[i]], ...)
  9. mask$eval_all_summarise(quo)
 10. dplyr (local) eval()
Error in summarise(., R_SQUARED = rsq_vec(strength, .pred), RMSE = rmse_vec(strength, : Caused by error in `rsq_vec()`: ! could not find function "rsq_vec"

**Comment by arkanayudhistira** [comment, marked as answer]
Dear @alexanderbudiman,

Mengacu dari keterangan error yang dihasilkan, sepertinya fungsi `rsq_vec` tidak ada di R, untuk menghitung nilai R-Squared, dapat diambil dengan fungsi `summary(model)$r.squared`. Untuk metrics lainnya, dapat menggunakan fungsi dari `library(MLmetrics)` yaitu `RMSE()`, `MAE()`, `MSE()`, dan `MAPE()`.
